jupyterhub/utils.py

```diff
@@ -956,6 +957,24 @@ def subdomain_hook_idna(name, domain, kind):
     return f"{safe_name}{suffix}.{domain}"
 
 
+# A URL safe identifier can be used in URL paths without escaping:
+# Unicode letters, digits, _, -
+_safe_unicode_identifier_re = re.compile(r'^[\w-]{1,63}$')
```

Comment:
What's a good size limit? If we want something that can be converted to a DNS safe string and back then we need to take account of unicode being escaped. Or we could stick with the KubeSpawner approach of using a hash if the string is too long?

Reply:
if we do that, the 1123 check in kubespawner is [here](https://github.com/jupyterhub/kubespawner/blob/0020c0b39e4e8c9b072f91b2875af18b6c8f57ee/kubespawner/slugs.py#L65).